parlai/mturk/webapp/server.py

```diff
@@ -304,16 +319,158 @@ def get(self, worker_target):
         self.write(json.dumps(data))
 
 
+class AssignmentHandler(BaseHandler):
+    def initialize(self, app):
+        self.state = app.state
+        self.subs = app.subs
+        self.sources = app.sources
+        self.port = app.port
+        self.data_handler = app.data_handler
+
+    def get(self, assignment_target):
+        # Extract assignment
+        assignments = [self.data_handler.get_assignment_data(
+            assignment_target)]
+        pairings = self.data_handler.get_pairings_for_assignment(
+            assignment_target)
+        processed_assignments = merge_assignments_with_pairings(
+            assignments, pairings, 'assignment {}'.format(assignment_target))
+        assignment = processed_assignments[0]
+
+        # Get assignment details to retrieve assignment content
+        run_id = assignment['run_id']
+        onboarding_id = assignment['onboarding_id']
+        conversation_id = assignment['conversation_id']
+        worker_id = assignment['worker_id']
+
+        onboard_data = None
+        if onboarding_id is not None:
+            onboard_data = MTurkDataHandler.get_conversation_data(
+                run_id, onboarding_id, worker_id, self.state['is_sandbox'])
+
+        assignment_content = {
+            'onboarding': onboard_data,
+            'task': MTurkDataHandler.get_conversation_data(
+                run_id, conversation_id, worker_id, self.state['is_sandbox']),
+        }
+
+        # Get assignment instruction html
+        taskname = '_'.join(run_id.split('_')[:-1])
+        find_location = 'parlai.mturk.tasks.{}.task_config'.format(taskname)
+        find_location_internal = \
+            'parlai_internal.mturk.tasks.{}.task_config'.format(taskname)
+        try:
```

Comment:
suppose you have two versions of a task, one internal and one external (e.g. you're extending an external task internally for a project), is there a way to specify overriding this order and using the internal task instead?

Reply:
Right now I haven't added any flag to do such a thing. I'm unsure how frequently we'd run into a problem like this though, as usually things that end up public aren't collecting any more data, and if you're making an extension on a task it'll have an extended name (usually).